src/components/labels/SSIStatusLabel/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { FC } from 'react'
+import { View, ViewStyle } from 'react-native'
+
+import { ConnectionStatusEnum, CredentialStatusEnum, IssuerStatusEnum, LabelStatus } from '../../../@types'
+import { translate } from '../../../localization/Localization'
+import { statuses } from '../../../styles/colors'
+import {
+  SSIStatusLabelBadgeContainer as BadgeContainer,
+  SSIStatusLabelContainerStyled as Container,
+  SSIStatusLabelStatusCaptionStyled as StatusCaption
+} from '../../../styles/components'
+import SSICheckmarkBadge from '../../badges/SSICheckmarkBadge'
+import SSIExclamationMarkBadge from '../../badges/SSIExclamationMarkBadge'
+
+export interface IProps {
+  status: LabelStatus
+  showIcon?: boolean
+  color?: string
+  style?: ViewStyle
+}
+
+const SSIStatusLabel: FC<IProps> = (props: IProps): JSX.Element => {
+  const { status, color = statuses[status], style, showIcon = false } = props
+
+  return (
+    <Container style={[style, { borderColor: color }]}>
+      {showIcon && <BadgeContainer>{getStatusBadge(status, color)}</BadgeContainer>}
+      <StatusCaption style={{ color }}>{getStatusTranslation(status)}</StatusCaption>
+    </Container>
+  )
+}
+
+const getStatusTranslation = (status: LabelStatus): string => {
+  switch (status) {
```

Comment:
Not sure about this, but let's keep it for now. We can always refactor these into separate methods

Reply:
I agree it is not the best option, but i wanted to merge all the status components we had as they were copies. And i need to stop refactoring stuff else this ticket will not get done. And i already did what we always said we should not do and that is pull in a lot of refactor work which is outside of the scope of the ticket...sorry :)